WrightTools/data/_axis.py

```diff
@@ -146,7 +146,10 @@ def variables(self):
         except (AssertionError, AttributeError):
             pattern = '|'.join(map(re.escape, operators))
             keys = re.split(pattern, self.expression)
-            indices = [self.parent.variable_names.index(key) for key in keys]
+            indices = []
+            for key in keys:
+                if key in self.parent.variable_names:
+                    indices.append(self.parent.variable_names.index(key))
```

Comment:
Could still be done with lost comprehension

Reply:
yeah but I think that this is more readable, to be honest---just a mild preference

I'll update to list comprehension if you want

Reply:
Fair enough, just bring it up as it's possible list comprehension maybe more performance (though these are cached anyway). Just something to think about/have in the back of our minds/on the record.